src/todoList.js

```diff
@@ -0,0 +1,346 @@
+
+    return (milliseconds1 < milliseconds2) ? -1 : (milliseconds1 > milliseconds2) ? 1 : 0;
+}
+
+function sortByStatus() {
+
+    let todoItems = savedTodoItems.items;
+
+    let openTodoItems = todoItems.filter((todoItem) => {
+        return todoItem.status !== statusItem.Done;
+    });
+
+    let doneTodoItems = todoItems.filter((todoItem) => {
+        return todoItem.status !== statusItem.Open;
+    });
+
+    todoItems = openTodoItems.concat(doneTodoItems);
+
+    setSortedItems(todoItems, typeOfSorting.Status);
+
+    let listItems = document.getElementById("todo-list");
+    listItems.innerHTML = ""; //drop all children
+
+    todoItems.forEach((todoItem) => {
+        addElementToDOM(todoItem);
+    });
+}
+
+function sortTodoItems(sortType) {
+
+    let todoItems = savedTodoItems.items;
+
+    if (sortType === typeOfSorting.Date) {
+        todoItems.sort(dateComparison);
+    }
+
+    if (sortType === typeOfSorting.Alphabetical) {
+        todoItems.sort(stringComparision);
+    }
+
+    setSortedItems(todoItems, sortType);
+
+    let listItems = document.getElementById("todo-list");
+    listItems.innerHTML = ""; //drop all children
+
+    todoItems.forEach((todoItem) => {
+        addElementToDOM(todoItem);
+    });
+}
+
+function setSortedItems(sortedItems, sortType) {
+
+    savedTodoItems.items = sortedItems;
+    savedTodoItems.sortState = sortType;
+    localStorage.setItem("todoItems", JSON.stringify(savedTodoItems));
+}
+
+function stringComparision(item1, item2) {
+
+    let textA = item1.description.toUpperCase();
+    let textB = item2.description.toUpperCase();
+
+    return (textA < textB) ? -1 : (textA > textB) ? 1 : 0;
+
+}
+
+function sortItems() {
+
+    let sortType = document.querySelector('#sorting-options').value;
+
+    if (sortType === typeOfSorting.Status) {
+        sortByStatus();
+    }
+    else {
+        sortTodoItems(sortType);
+    }
+}
+
+function addItemToList(e) {
+
+    if (e) {
+        //Get the Unicode value of the pressed keyboard key
+        //using which or keyCode depends on browser support
+        let code = e.keyCode ? e.keyCode : e.which;
+        const Enter = 13;
+
+        if (code === Enter) {
+
+            const itemDescription = document.getElementById("task-description").value;
+            const sortOption = document.getElementById("sorting-options").value;
+            let newItemToSave = {
+                description: itemDescription,
+                date: new Date(),
+                status: 'open',
+                id: savedTodoItems.counter
+            };
+
+            const isNew = true;
+            let indexToInsert = getIndexToInsert(sortOption, newItemToSave);
+
+            debugger;
+            saveStateToLocalStorage(newItemToSave, sortOption, indexToInsert);
+            addElementToDOM(newItemToSave, isNew, indexToInsert);
+            clearTextInput();
+        }
+    }
+}
+
+
+function addElementToDOM(newItemToSave, isNew = false, indexToInsert) {
```

Comment:
No need for 'isNew' parameter anymore.
You can just call it with the index of the current element:
todoItems.forEach((todoItem, index) => {
   addElementToDOM(todoItem, index);
});